src/main/java/edu/ie3/netpad/io/controller/IoController.java

```diff
@@ -75,56 +85,242 @@ public void registerGridControllerListener(ChangeListener<IOEvent> listener) {
     return sampleGridOpt;
   }
 
-  public Optional<JointGridContainer> loadGridFromCsv(File absoluteFilePath, String csvSeparator) {
-
-    // get the CsvGridSource
-    String gridName =
-        absoluteFilePath.getAbsolutePath()
-            .split(File.separatorChar == '\\' ? "\\\\" : File.separator)[
-            absoluteFilePath
-                    .getAbsolutePath()
-                    .split(File.separatorChar == '\\' ? "\\\\" : File.separator)
-                    .length
-                - 1];
-
-    // get grid and inform listeners if present
-    return new CsvGridSource(absoluteFilePath.getAbsolutePath(), gridName, csvSeparator)
-        .getGrid()
-        .map(
-            grid -> {
-              notifyListener(new ReadGridEvent(grid));
-              return grid;
-            });
+  public boolean loadGridFromArchive(
+      File absoluteArchivePath,
+      String csvSeparator,
+      IoDialogs.CsvIoData.DirectoryHierarchy hierarchy) {
+    /* Create a temp directory */
+    Path tmpDirectory;
```

Comment:
There is no need to extract the files of the tarball as you can read data from. This would avoid the overhead of creating a tmp directory as well as making it more bulletproof. See https://www.daniweb.com/programming/software-development/threads/400615/reading-data-within-tar-files for an example.

Reply:
![not bad](https://media.giphy.com/media/5pUHtgG70JjO9YHZPD/source.gif)

Reply:
Please let us postpone this feature and let it succeed PowerSystemDataModel#234. From my first glance, the needed adaptions are definitely better located in PowerSystemDataModel or PowerSystemUtils.